evm/src/cpu/kernel/asm/core/call.asm

```diff
@@ -212,3 +222,34 @@ global after_call_instruction:
     EXIT_KERNEL
     // (Old context) stack: new_ctx
 %endmacro
+
+%macro copy_mem_to_calldata
+    GET_CONTEXT
+    %stack (ctx, new_ctx, args_offset, args_size, new_ctx) ->
```

Comment:
Hm why take two copies of `new_ctx`? I actually didn't know `%stack` would accept it, maybe we should change it later to require unique names.

Reply:
Oh good catch, I cleaned that up. I agree we should probably throw an error when there are duplicate names.